grype/matcher/rpm/rhel_eus.go

```diff
@@ -0,0 +1,326 @@
+func processVulnerabilityWithAdvisories(v *version.Version, disclosure vulnerability.Vulnerability, advisoryOverlays []result.Result) (*vulnerability.Vulnerability, match.Details) {
+	fixVersions := version.NewSet(true)
+	var constraints []version.Constraint
+	var state vulnerability.FixState
+	var allAdvisoryDetails match.Details
+
+	// check if we're vulnerable to the original disclosure
+	if isVulnerableVersion(v, disclosure.Constraint, disclosure.ID) {
+		constraints = append(constraints, disclosure.Constraint)
+	}
+
+	// process advisory overlays, incorporating new fix versions and updating the version constraints
+	for _, advisoryOverlay := range advisoryOverlays {
+		allAdvisoryDetails = append(allAdvisoryDetails, advisoryOverlay.Details...)
+		processAdvisoryVulnerabilities(v, advisoryOverlay.Vulnerabilities, fixVersions, &constraints, &state)
+	}
+
+	if len(constraints) == 0 {
+		// all of the advisories showed we're not vulnerable, so we can skip this disclosure
+		return nil, nil
+	}
+
+	patchedRecord := buildPatchedVulnerabilityRecord(v, disclosure, fixVersions, constraints, state)
+	return &patchedRecord, allAdvisoryDetails
+}
+
+// processAdvisoryVulnerabilities processes vulnerabilities from advisory overlays, applying any new fix versions and updating the given fix state / constraints.
+func processAdvisoryVulnerabilities(v *version.Version, advisories []vulnerability.Vulnerability, fixVersions *version.Set, constraints *[]version.Constraint, state *vulnerability.FixState) {
+	for _, advisory := range advisories {
+		if advisory.Fix.State == vulnerability.FixStateWontFix && *state != vulnerability.FixStateFixed {
+			*state = advisory.Fix.State
+		}
+
+		applicableFixes := neededFixes(v, advisory.Fix.Versions, advisory.Constraint.Format(), advisory.ID)
+		if len(applicableFixes) == 0 {
+			// none of the fixes on this advisory are greater than the current version, so we can skip this advisory
+			continue
+		}
+
+		// we're vulnerable! keep any fix versions that could have been applied
+		*constraints = append(*constraints, advisory.Constraint)
+		fixVersions.Add(applicableFixes...)
+		if *state != vulnerability.FixStateFixed {
+			*state = advisory.Fix.State
+		}
+	}
+}
+
+// buildPatchedVulnerabilityRecord creates the final patched vulnerability record from the original disclosure and fix/constraint information from applicable advisories.
+func buildPatchedVulnerabilityRecord(v *version.Version, disclosure vulnerability.Vulnerability, fixVersions *version.Set, constraints []version.Constraint, state vulnerability.FixState) vulnerability.Vulnerability {
+	patchedRecord := disclosure
+
+	if state == vulnerability.FixStateFixed {
+		patchedRecord.Fix.Versions = nil
+		for _, fixVersion := range fixVersions.Values() {
+			patchedRecord.Fix.Versions = append(patchedRecord.Fix.Versions, fixVersion.Raw)
+			fixConstraint, err := version.GetConstraint(fmt.Sprintf("< %s", fixVersion), v.Format)
```

Comment:
This `fixVersion` is causing the `.String()` to be called which fails to parse due to parenthesis and some entries to be excluded, I think it should be `fixVersions.Raw` here